dali/sasha/saxref.cpp

```diff
@@ -340,8 +357,9 @@ struct cDirDesc
 
         // NB: Creation only happens during scanDirectories, this function should be called from [thread-safe] lookupDir()
         // lookupDirNonThreadSafe is also called in findDirectory during scanLogicalFiles. It will not reach here, and does not need to be thread safe )
-        cDirDesc *ret = new cDirDesc(name);
-        dirs[name] = std::unique_ptr<cDirDesc>(ret);
+        auto uniqueDirPtr = cDirDesc::create(name);
+        cDirDesc *ret = uniqueDirPtr.get();
+        dirs[name] = std::move(uniqueDirPtr);
```

Comment:
question: is this better than the create methods returning a plain cDirDesc pointer, and the code here looking like this:
```
cDirDesc *ret = cDirDesc::create(name);
dirs.emplace(name, ret);
```

passing back unique_ptr's may be fine, better, if there's little to no copying overheads. I suspect that's true with move operator semantics.

I wouldn't change it, just thinking out aloud.